### ISSUE TITLE
Training RMSE/R2 not optimized after 50 epochs?

### ISSUE THREAD

**Issue by hmunozb** [SysBioChalmers/DLKcat#3]
I attempted to train the DL model on the included Data/input dataset, but the RMSE/R2 performance for the optimal hyperparameters does not seem to improve well within 50 epochs. 

The training command I am  using in `Code/model` is
```
# Conda environment: python=3.8.13=h12debd9_0, pytorch=1.11.0=py3.8_cuda11.3_cudnn8.2.0_0, 
#   numpy=1.22.3=py38h99721a1_2
                    #dataset   radius  ngram   dim     layer_gnn   window   layer_cnn  layer_output   lr   lr_decay   decay_interval  weight_decay  iteration  setting
python -u run_model.py  all      2      3     20            3       11         3           3        1e-3     0.5            10          1e-6           50         test
```

My MAEs output file using the same seed in the source file is
```tsv
Epoch	Time(sec)	RMSE_train	R2_train	MAE_dev	MAE_test	RMSE_dev	RMSE_test	R2_dev	R2_test
1	102.55310953292064	1.4312380283147441	0.09236332967789695	1.125049116795749	1.1250578585008904	1.4471744973977028	1.4677463388764231	0.060686416041210056	0.04520540426306441
2	203.64777798799332	1.362037256540892	0.17801049164963223	1.0671505529348142	1.058187257840833	1.3639318885468796	1.366977607844821	0.16563866907561975	0.1718085517220319
3	307.93242862902116	1.3323272460802105	0.2134793698365396	1.0579200712753927	1.0571677217552549	1.35433410797124	1.3544796493888676	0.17733990055978488	0.18688324676111367
4	417.40169664192945	1.31506452348586	0.23372893985948295	1.0468607239454852	1.0368631535890154	1.3480473355980844	1.346872249878544	0.184959693811926	0.19599129511149382
....
199	19898.319774403004	1.1893049201013368	0.3732783018122712	1.014455404012681	1.0182008302795162	1.3319525524402698	1.3474205021762213	0.20430558351836126	0.19533660891554494
200	19997.01809749799	1.189304904289939	0.3732783184763676	1.0144553978328625	1.018200828574738	1.3319525447368734	1.3474204932245593	0.2043055927222167	0.19533661960719573
```

However, the training R2 reaches 0.86 by the 50th epoch in the original MAEs output, and should even perform better by the second epoch.  Are the hyperparameters I am running being set correctly?

**Comment by feiranl** [comment]
Could you print out your installed package by running` conda list `or` pip list`?

**Comment by feiranl** [comment]
We have tested it in pytorch = 1.4.0 and 1.8.0. Could this be the reason? I will test again today in pytorch 1.11.0, will let you know.

**Comment by feiranl** [comment]
I created a new environment with pytorch=1.11.0 today,  I tested on my computer(cpu), it works fine.  I am using that on gpu now.
```
pip install torch==1.11.0          
pip install scikit-learn==0.23.2
pip install Biopython==1.78
pip install rdkit-pypi
pip install seaborn==0.11.0
pip install Matplotlib==3.3.2
pip install pandas==1.1.3
pip install SciPy==1.5.2
pip install NumPy==1.20.2
```
```tsv
Training...
Epoch	Time(sec)	RMSE_train	R2_train	MAE_dev	MAE_test	RMSE_dev	RMSE_test	R2_dev	R2_test
1	1850.890769584	1.3438182532736704	0.1998537546862883	0.9035335875736547	0.9133727147194339	1.1992452517734593	1.2359207287270122	0.3549627178018241	0.32299921362051043
2	3714.328295292	1.084357741076502	0.4790051045211742	0.8155829147555056	0.827786833740544	1.0718060717840199	1.1102980169965346	0.4847698989405582	0.4536297183959668
3	5565.796247834	1.0039601096588002	0.5533974334491122	0.7873367654527741	0.8069042074566374	1.046123391943784	1.0935254266866503	0.509166011683243	0.4700123957534842
4	7421.475990792	0.9586988949066316	0.5927578201651297	0.7794618119884426	0.7914288478966054	1.0353950991924916	1.084159005412381	0.5191816743983877	0.4790525652715526
5	9266.206740167	0.9310297750368882	0.6159255295480108	0.7454487619816593	0.7643532149817572	1.011139254568574	1.0703361718545579	0.5414457282064841	0.4922518545061749
6	11125.5231325	0.9118198494526261	0.631611228691773	0.790947889391016	0.8055422126099699	1.0413691513693535	1.0933207478520355	0.5136171889336283	0.470210776313323
7	12984.743249333	0.8906566336945769	0.6485132858747704	0.7597130137872721	0.7625249781298686	1.0223852634093529	1.0676801557246463	0.5311888160688671	0.4947686600756299
8	14841.661516167	0.8750188583259594	0.6607474459657958	0.7765459976505641	0.7862476204838971	1.0301065805234337	1.0810044343642546	0.5240809108014289	0.48207975010564863
9	16692.103153875	0.8614303673156085	0.6712023877816913	0.7394888517778917	0.7643921505036454	1.0224698312506153	1.0881460149657793	0.5311112562860693	0.4752139373068053
10	18538.492715834	0.790652954533467	0.7230125473773831	0.7279709099352978	0.7501985592397513	1.0082084445331096	1.0707727209979077	0.5441001355827655	0.4918375879860142
```
